components/logins/ffi/src/lib.rs

```diff
@@ -14,9 +14,13 @@ use ffi_support::{
 };
 use logins::{Login, PasswordEngine, Result};
 use std::os::raw::c_char;
+use std::sync::{Arc, Mutex};
 
 lazy_static::lazy_static! {
-    static ref ENGINES: ConcurrentHandleMap<PasswordEngine> = ConcurrentHandleMap::new();
+    // TODO: this is basically a RwLock<HandleMap<Mutex<Arc<Mutex<...>>>>.
+    // but could just be a `RwLock<HandleMap<Arc<Mutex<...>>>>`.
+    // Find a way to express this cleanly in ffi_support?
```

Comment:
Yes, that's why we need it. We don't really need to worry about deadlocks in practice since there's no way to acquire the locks in a different order than the one we currently have.